Spigot-Server-Patches/0316-Add-config-to-disable-ender-dragon-legacy-check.patch

```diff
@@ -0,0 +1,60 @@
+From 755f37848580b8f968e226950e2a6157e7bdd6aa Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Fri, 22 Jun 2018 10:38:31 -0500
+Subject: [PATCH] Add config to disable ender dragon legacy check
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 99ad40fa3..06c54690f 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -524,4 +524,9 @@ public class PaperWorldConfig {
+     private void shieldBlockingDelay() {
+         shieldBlockingDelay = getInt("game-mechanics.shield-blocking-delay", 5);
+     }
++
++    public boolean scanForLegacyEnderDragon = true;
++    private void scanForLegacyEnderDragon() {
++        scanForLegacyEnderDragon = getBoolean("game-mechanics.scan-for-legacy-ender-dragon", true);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/EnderDragonBattle.java b/src/main/java/net/minecraft/server/EnderDragonBattle.java
+index ce0dcbc68..a8c587e30 100644
+--- a/src/main/java/net/minecraft/server/EnderDragonBattle.java
++++ b/src/main/java/net/minecraft/server/EnderDragonBattle.java
+@@ -42,7 +42,12 @@ public class EnderDragonBattle {
+     public EnderDragonBattle(WorldServer worldserver, NBTTagCompound nbttagcompound) {
+         this.c = (BossBattleServer) (new BossBattleServer(new ChatMessage("entity.EnderDragon.name", new Object[0]), BossBattle.BarColor.PINK, BossBattle.BarStyle.PROGRESS)).setPlayMusic(true).c(true);
+         this.e = Lists.newArrayList();
+-        this.n = true;
++        // Paper start
++        setScanForLegacyFight(worldserver.paperConfig.scanForLegacyEnderDragon);
++        if (!scanForLegacyFight()) {
++            setDragonKilled(true);
++        }
++        // Paper end
+         this.d = worldserver;
+         if (nbttagcompound.hasKeyOfType("DragonKilled", 99)) {
+             if (nbttagcompound.b("DragonUUID")) {
+@@ -499,4 +504,18 @@ public class EnderDragonBattle {
+         }
+ 
+     }
++
++    // Paper start - OBFHELPERS
++    private void setDragonKilled(boolean dragonKilled) {
```

Comment:
This should be right above the `k` field.

Reply:
sorry, new to this concept of "obf helpers" - will fix